include/r3.h

```diff
@@ -46,7 +46,7 @@ typedef struct _node R3Node;
 typedef struct _R3Route R3Route;
 
 struct _node  {
-    R3Edge * edges;
```

Comment:
The edges was pointed to an array of edge, which is a continuos memory fragment. the changes from you is to making each edge in different memory fragment, to get each edge, there is an indirection fetching. (which can't be done by simple indexing)